Closure_Front_End/src/views/Home.vue

```diff
@@ -112,20 +163,69 @@ export default {
         this.track = response.data.results[0]
       })
     }
+    /** if the user is loged in, then fetching his data from DB, else doing nothing */
+    // if (this.$auth.isAuthenticated.value) {
+    //   this.getUsername().then(function (username) {
+    //     axios
+    //       .get("http://127.0.0.1:8000/api/v1/students/?search=" + username, {
+    //         headers: {
+    //           Authorization: "Token 0782d1d5118827d8f32cdeaddde60a8bb53d7625",
+    //         },
+    //       })
+    //       .then((response) => (this.student = response))
+    //       .then((response) => this.createAllCourses(response));
+    //   });
+    // }
   },
+
   methods: {
+    getUsername() {
+      return this.$auth
+        .getIdTokenClaims()
+        .then((response) => response.nickname);
+    },
+
+    /** fetch all student's courses from the DB and store them in allcourses */
+    createAllCourses(student) {
+      for (course in student.courses) {
```

Comment:
I recommend replacing this line by
`for (const course of student.courses)`

- It is not recommended to create a variable without the `let`/`const` keyword, this is probably why there are many warnings below
- Generally I would avoid looping via the `in` operator, since it might not always do what you want, it doesn't only iterate over iteration elements(like `for of`) but also "enumerable properties"